src/server/middleware/twitter.middleware.ts

```diff
@@ -286,7 +285,7 @@ function loadTwitterData(
 
   const twitterApiRequest: TwitterApiRequest = {
     query: `(@${user} OR url:twitter.com/${user}) -from:${user} -is:retweet`,
-    maxResults: BATCH_SIZE,
+    maxResults: ENTERPRISE_BATCH_SIZE,
```

Comment:
Isn't choice of batch size determined by `credentials.useEssentialOrElevatedV2`?

Reply:
Yes, @kilemensi. But this function (`loadTwitterData`) will only be called if using enterprise api, so no need to check if it is the `essential` api

Reply:
and loadTwitterDataV2 will always be called in essential mode @kelvinkipruto? we're hard-coding batch size to [100](https://github.com/CodeForAfrica/harassment-manager/blob/3ab9e2b07f0bc49674cb8facb41237b532df4f9e/src/server/middleware/twitter.middleware.ts#L334) even when doing full search.

Reply:
@kilemensi Even for full search, the max batch size is 100

Reply:
@kelvinkipruto don't answer for the sake of answering... you're taking my time away from other more pressing issues for something **_you_** can look up. Here is the [line](https://github.com/conversationai/harassment-manager/blob/6c9a5012589423a6e030d802d4086c60225f7e77/src/server/middleware/twitter.middleware.ts#L333) in the original `loadTwitterDataV2` which is using `BATCH_SIZE` define as `500`.

Where is the information that full search must be limited to 100 coming from; if there is, then please include it as a comment in the code so that there is no need for this back and forth.

Reply:
@kilemensi Sorry for the confusion . This information was shared on the [migration docs](https://docs.google.com/document/d/1X17DsnkltFJBWbm2YfTCY4uap35gqQDe_rXDIAKq4eA/edit?resourcekey=0-Z_RKjm0UvWB_nM4gJspcIw#heading=h.d0mx998uxrqt) shared with us. Point No 2.
I couldn't find the documentation for it, but when I use 500 batch size, it returns an error saying max batch size is 100

Reply:
In a nutshell, only enterprise has batch size 500, all the rest is 100

Reply:
Straight from the doc you linked to @kelvinkipruto:

> Summary: Again, a one-line change. Academic v2 supports up to 500 results per request, but Essential/Elevated support only 100.

Sounds like only Essential/Elevated supports up to 100 but the rest supports more.